### PR TITLE
Editor: Display dialog if lacking upload permissions

### LINKED ISSUE #2481: Use dialog to warn users about lack of upload permissions
## Feature Description

This is a follow up to #1372, specifically https://github.com/google/web-stories-wp/issues/1372#issuecomment-642446562

As per the design at https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=9631%3A10:

If as user does not have capabilities to upload media, display a dialog saying "You don't have access to upload media. Check with your administrator to request access."

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* If as user does not have capabilities to upload media, display a dialog saying "You don't have access to upload media. Check with your administrator to request access."

### PR DESCRIPTION
**Pull request by zidianlyu**
Adds a dialog to display the access restriction warning message.

## Summary

Add a popup dialog to notify the user to check access permission when they are restricted to upload.

## Relevant Technical Choices

1. Add a RestrictAccessDialog component
2. Add a state [hasRestrictAccess] and pass MissingUploadPermissionDialog component under mediaPane component.
3. Add a onRestrictAccess props function on useMediaPicker, for controlling the dialog show.

## To-do

N/A

## User-facing changes

Will get a popup warning dialog instead of Snackbar when they click on the upload button and do not have access.

## Testing Instructions

1. The Snackbar should no longer shows up.
2. The dialog should popup with the accurate instruction test when not permitted user click on the 'Upload' button.
3. The dialog should close when the user click on 'GOT IT'

---

Fixes #2481

### CODE REVIEW

**Comment by swissspidy** [comment]
Same needs to be done to `useUploader` I presume?

File: assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Plain } from '../../../../button';
+import Dialog from '../../../../dialog';
+
+function RestrictAccessDialog({ open, onClose }) {
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js]: A storybook story would be nice for this dialog, as we have stories for all other dialogs already.

**Comment by dreamofabear** [comment]
Looks good! Please add a screenshot so PM/UX can review.

File: assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js (hunk 1 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js]: Added story book.

**Comment by zidianlyu** [comment]
> Looks good! Please add a screenshot so PM/UX can review.

Attach a screenshot for thee result.
Screen Shot 2020-11-09 at 10 52 56 PM

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js [hunk 2]
```diff
@@ -121,6 +122,10 @@ function MediaPane(props) {
     insertElement: state.actions.insertElement,
   }));
 
+  const [showRestrictAccessDialog, setShowRestrictAcccessDialog] = useState(
```
Comment (barklund) [inline assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: That's a lot of c's! Maybe too many?

```suggestion
  const [showRestrictAccessDialog, setShowRestrictAccessDialog] = useState(
```

Secondarily, I prefer the naming style of an `is` or `has` prefix for booleans, but I'm not that rigid on it (anymore), so this is probably fine.

File: assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js (hunk 1 above)
Comment (barklund) [inline assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js]: I'm not sure about the naming of this component (and related properties). It's not a component to "restrict access", but rather a component to inform of "access restriction".

**Review by barklund — APPROVED** [review]
Works like a charm - just a few naming concerns.

File: assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js (hunk 1 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js]: MissingPermissionsDialog?

File: assets/src/edit-story/components/library/panes/media/local/stories/restrictAccessDialog.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      actions: { deleteMediaElement: action('close dialog') },
+    },
+  };
+  const snackbarValue = { showSnackbar: action('show snackbar') };
+
+  return (
+    <SnackbarContext.Provider value={snackbarValue}>
+      <MediaContext.Provider value={mediaValue}>
+        <ApiContext.Provider value={apiValue}>
+          <RestrictAccesssDialog open={true} onClose={action('closed')} />
```
Comment (dreamofabear) [inline assets/src/edit-story/components/library/panes/media/local/stories/restrictAccessDialog.js]: Too many S's.
```suggestion
          <RestrictAccessDialog open={true} onClose={action('closed')} />
```

File: assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js]: `MissingUploadPermissionsDialog` to make it clear which kind of permission?

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js (hunk 2 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: Accepted, done!

File: assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js (hunk 1 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/library/panes/media/local/restrictAccessDialog.js]: Updated as `MissingUploadPermissionsDialog`

File: assets/src/edit-story/components/library/panes/media/local/stories/restrictAccessDialog.js (hunk 3 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/library/panes/media/local/stories/restrictAccessDialog.js]: Updated is as `MissingUploadPermissionDialog`

**Pull request closed without merging**